addons/sale/models/sale_order.py

```diff
@@ -1245,6 +1245,25 @@ def get_portal_last_transaction(self):
         self.ensure_one()
         return self.transaction_ids._get_last()
 
+    def _get_order_lines_to_report(self):
+
+        def show_line(line):
+            if line.product_id and not line.is_downpayment:
+                return True
+            else:
+                return line._get_downpayment_state() not in ["draft", "cancel"] and line._show_downpayment_section()
+
+        return self.order_line.filtered(show_line)
+
+    def _count_downpayment_lines(self):
+        return len(self.order_line.filtered(
+            lambda line: (
+                line.is_downpayment and  # it is indeed a downpayment
+                not line.display_type and  # and is not the downpayment section
+                line._get_downpayment_state() not in ["draft", "cancel"]  # and not on cancel or draft state
```

Comment:
don't add 'useless' comments please, only the 'is not the downpayment section' says something more than reading the code.